src/sentry/issues/suspect_flags.py

```diff
@@ -0,0 +1,126 @@
+from collections import defaultdict
+from collections.abc import Mapping
+from datetime import datetime
+
+from snuba_sdk import Column, Condition, Entity, Function, Op, Query, Request
+
+from sentry.seer.workflows.compare import kl_compare_sets
+from sentry.utils.snuba import raw_snql_query
+
+Attributes = Mapping[str, dict[str, float]]
+AttributesRow = tuple[str, str, float]
+Score = tuple[str, float]
+
+
+def get_suspect_flag_scores(
+    org_id: int, project_id: int, start: datetime, end: datetime, primary_hash: str
+) -> list[Score]:
+    """
+    Queries the baseline and outliers sets. Computes the KL scores of each and returns a sorted
+    list of key, score values.
+    """
+    baseline_rows = query_flag_rows(org_id, project_id, start, end, primary_hash=None)
+    baseline = as_attribute_dict(baseline_rows)
+
+    outliers_rows = query_flag_rows(org_id, project_id, start, end, primary_hash=primary_hash)
+    outliers = as_attribute_dict(outliers_rows)
+
+    return kl_score(baseline, outliers)
+
+
+def query_flag_rows(
+    organization_id: int,
+    project_id: int,
+    start: datetime,
+    end: datetime,
```

Comment:
What kinds of date ranges do we expect to run this query on?

Reply:
Presumably up to 90 days.  Is that our max retention for errors?

Reply:
That's the max retention. I don't have a great understanding of this what this project is doing as a whole - but scanning all 90 days of data can be expensive, so if there are ways to limit it then it'd be helpful

Reply:
Yeah.  I imagine larger projects would degrade and we would need to optimize as we understood the problem better.  The most likely outcome is applying a sampling argument to the query so we scan fewer rows.